test/clients/azure/osa.go

```diff
@@ -24,3 +38,41 @@ func (cli *Client) OSAResourceGroup(ctx context.Context, resourcegroup, name, lo
 	rgid := *app.ApplicationProperties.ManagedResourceGroupID
 	return rgid[strings.LastIndexByte(rgid, '/')+1:], nil
 }
+
+// UpdateOSACluster updates an OpenshiftManagedCluster by providing both the currently applied external manifest and
+// an updated internal manifest
+// NOTE: This method will go away as soon as the admin API is ready. Until then, only the KeyRotation or other e2e tests
+// needing admin functionality should call this method.
+func (cli *Client) UpdateOSACluster(ctx context.Context, external *v20180930preview.OpenShiftManagedCluster, config *api.PluginConfig) (*v20180930preview.OpenShiftManagedCluster, error) {
+	// Remove the provisioning state before updating, if set
+	external.Properties.ProvisioningState = ""
+
+	// create logger
+	logrus.SetLevel(log.SanitizeLogLevel("Debug"))
+	logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+	logger := logrus.NewEntry(logrus.StandardLogger())
+
+	var oc *v20180930preview.OpenShiftManagedCluster
+	var err error
+	// simulate the API call to the RP
+	if err := wait.PollImmediate(5*time.Second, 1*time.Hour, func() (bool, error) {
+		if oc, err = fakerp.CreateOrUpdate(ctx, external, logger, config); err != nil {
+			if autoRestErr, ok := err.(autorest.DetailedError); ok {
+				if urlErr, ok := autoRestErr.Original.(*url.Error); ok {
+					if netErr, ok := urlErr.Err.(*net.OpError); ok {
+						if sysErr, ok := netErr.Err.(*os.SyscallError); ok {
+							if sysErr.Err == syscall.ECONNREFUSED {
+								return false, nil
```

Comment:
Do you actually see this happening in the test context?  ie is this code actually needed or is it superstition? ;-)

Reply:
We were seeing this in the fake RP create case. Probably not needed for updates since we know the RP is already running.

Reply:
I lifted this bit of code from cmd/createorupdate.go but it is indeed not needed. I'll take it out